### ISSUE TITLE
Navigation doesn't work properly in the plugins involves Tree

### ISSUE THREAD

**Issue by mmelko** [hawtio/hawtio-next#903]
This might be caused by resolution of the path, where selected mbeans aren't included into the path, so i.e source of the route1 is translated to `[host]/hawtio/camel/source`. So when we paste this link to the browser we are automatically redirected to the root of the camel plugin with the context table. 

Similar behaviour when we selected different mbean of the same type. Instead of perceive the selected tab it's automatically redirected to the first tab.

**Comment by tadayosi** [comment]
@mmelko Look at how Hawtio v2 did for the persistent navigation. I remember it used query parameters such as `?nid=` for keeping the current mbean location. That said, I don't think using query params is a good design in 2024. So the first thing we should do would be to design how we encode the path location into URL. Do you have any ideas?

**Comment by tadayosi** [comment]
I wouldn't say it's a bug so let me change it to enhancement.

**Comment by mmelko** [comment]
Sure, I marked is a bug because the functionality is lacking compared to v2. 

My idea from top of my head is to use path param and mbean names (i.e `[host]/hawtio/camel/source/routes`) That would require extending either current context or create new with something like searchNode and then obtain the objectName.

**Comment by tadayosi** [comment]
@mmelko Hmm, then how do we encode the selected node into the path?  Should we still use a query parameter for the node selection?

**Comment by mmelko** [comment]
so bad example .. routes would be the name of the mbean (not the object name). so for route1 the path would look like `[host]/hawtio/camel/source/route1`  That's how I'd like to see it.

**Comment by tadayosi** [comment]
@joshiraezcode I don't think so. You are free to investigate it but be aware that it's really a complicated one to implement correctly. I'd propose to first investigate it and write some solution plan for review before working on an actual fix.

**Comment by pseudaverse** [comment]
> [@mmelko](https://github.com/mmelko) Look at how Hawtio v2 did for the persistent navigation. I remember it used query parameters such as `?nid=` for keeping the current mbean location. That said, I don't think using query params is a good design in 2024. So the first thing we should do would be to design how we encode the path location into URL. Do you have any ideas?

@tadayosi This thing was quite useful in hawtio v2, with it and  "con=" query param we could share link to exact mbean/etc.. I guess the only way to do so is to add some state/location storing around JmxTreeView, and nid query param looks like a straightforward solution in this case. Why do you think its a bad idea? I would implement this logic (maybe firstly under some config param in hawtconfig.json/whatever), what do you think?

**Comment by tadayosi** [comment]
While #1967 is still not ready for merge, the majority of the work regarding this issue is already done. Let's close it.
